concrete-benchmark/src/utils.rs

```diff
@@ -0,0 +1,58 @@
+//! A module containing some utilities.
+
+/// This macro creates a string that represent the headline of the impl block being benchmarked.
+/// It type checks the input such that there is no risk of misspelling. Though the error triggered
+/// may be difficult to interpret ¯\_(ツ)_/¯.
+macro_rules! benchmark_name {
+    (impl $Trait:ident<$($Param:ty),+> for $Engine:ty) => {
+        {
+            let _type_check_signature: Box<dyn $Trait<$($Param),*, EngineError=<$Engine>::EngineError>> = Box::new(<$Engine>::new().unwrap());
+            let mut output = String::from("impl ");
+            output.push_str(format!("{}<", stringify!($Trait)).as_str());
+            $(
+            output.push_str(format!("{},", $crate::utils::type_name::<$Param>()).as_str());
+            )*
+            output.pop();
+            output.push_str(format!("> for {}", $crate::utils::type_name::<$Engine>()).as_str());
+            output
+        }
+    };
+}
+pub(crate) use benchmark_name;
+
+/// A function returning the name of the type (just the name, not the path).
+pub fn type_name<T>() -> &'static str {
+    std::any::type_name::<T>()
+        .split("::")
+        .collect::<Vec<_>>()
+        .pop()
+        .unwrap()
+}
+
+/// A trait to create values and vec of values of integer numeric types.
+pub trait RawNumeric: Sized {
```

Comment:
Here maybe we want to change `Raw` to fit the new naming in concrete-core (if there's one)?

Reply:
In core we did `Raw` -> `Value`. What about using `IntegerValue`  (the types here are integers for now).

Reply:
Yes `IntegerValue` sounds good.